rust-runtime/aws-smithy-json/src/deserialize/token.rs

```diff
@@ -578,6 +591,21 @@ pub mod test {
             Ok(None),
             expect_timestamp_or_null(value_null(0), Format::HttpDate)
         );
+        for &invalid in &[f64::INFINITY, f64::NEG_INFINITY, f64::NAN] {
+            assert_eq!(
+                Err(Error::new(
+                    ErrorReason::Custom(Cow::Owned(format!(
+                        "NaN and Infinity are not valid epochs: {}",
+                        invalid
+                    ))),
+                    None,
+                )),
+                expect_timestamp_or_null(
+                    value_number(0, Number::Float(invalid)),
```

Comment:
`NaN` and `Infinity` will always come in as string tokens since JSON doesn't support them as numbers, so this test isn't quite accurate.

Instead, you should iterate over `["NaN", "Infinity", and "-Infinity"]`, and use `value_string`.